### PR TITLE
use fetchStatus to fetch status of more than one jid

### PR DESCRIPTION
**Pull request by vinikjkkj**

### CODE REVIEW

File: src/Socket/chats.ts [hunk 1]
```diff
@@ -190,18 +190,20 @@ export const makeChatsSocket = (config: SocketConfig) => {
...
+			list,
 			{ tag: 'status', attrs: {} }
 		)
-		if(result) {
-			const status = getBinaryNodeChild(result, 'status')
+		return results.map(item => {
+			const status = getBinaryNodeChild(item, 'status')
 			return {
+				user: item.attrs.jid,
 				status: status?.content!.toString(),
```
Comment (azudindaem) [inline src/Socket/chats.ts]:
```ts
status: status && status.content ? status.content.toString() : null,
```

prevent undefined error if contact doesn't have status

additional info in readme need to change to

```ts
const jid = ["abx@s.whatsapp.net","xyz@s.whatsapp.net]
const status = await sock.fetchStatus(...jid)
console.log("status: " + status)
```

Not sure if we need this, but don't have any problem if others agree

**Review by azudindaem — CHANGES_REQUESTED** [review]

File: src/Socket/chats.ts (hunk 1 above)
Comment (vinikjkkj) [reply src/Socket/chats.ts]: i didn't change the original code in this part, I don't think that's necessary, the operator [ ! ] already do this

Comment (azudindaem) [reply src/Socket/chats.ts]: > i didn't change the original code in this part, I don't think that's necessary, the operator [ ! ] already do this

the current code (old) indeed has a problem if the status is undefined. It is better for us to handle this issue in this pr

status: status?.content!.toString() will throw an error

```txt
TypeError: Cannot read properties of undefined (reading 'toString')
    at /home/node_modules/@whiskeysockets/baileys/lib/Socket/chats.js:170:88
    at Array.map (<anonymous>)
    at Object.fetchStatus (/home/node_modules/@whiskeysockets/baileys/lib/Socket/chats.js:166:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
    
My suggestion is to safely handles cases where status or status.content might be undefined or null. we should returns null for the status property if status or status.content is not available..

Comment (vinikjkkj) [reply src/Socket/chats.ts]: applied

**Review by azudindaem — APPROVED** [review]
LGTM

**Comment by PurpShell** [comment]
breaking change

**Pull request closed without merging**